### PR TITLE
Move ImportDepsChecker to a checked in deploy jar.

### PR DESCRIPTION
**Pull request by iirina**
instead of embedding the deploy jar that is built when the bazel binary is built.

This comes in preparation for moving the Java tools to remote repositories (see #6316). This checked in jar will be moved to a remote repository and referenced from there.

Moving the Java tools happens in two phases:
1. Using checked-in deploy jars/binaries instead of those built when bazel was built
2. Moving the checked-in jars/binaries to a remote repository and make bazel reference them from there. 

This PR addresses the first point for `ImportDepsChecker`. Doing the move incrementally this way is safer and allows finding bugs earlier. It also allows moving the Java tools code base out of the bazel repository.

We are making this change gradually for each java tool defined in JAVA_TOOLS (found in src/BUILD).

### CODE REVIEW

File: third_party/java/java_tools/README.md [hunk 1]
```diff
@@ -17,3 +17,8 @@ third_party/java/java_tools/JacocoCoverage_jarjar_deploy.jar
 third_party/java/java_tools/turbine_deploy.jar
 third_party/java/java_tools/turbine_direct_binary_deploy.jar
 third_party/java/java_tools/SingleJar_deploy.jar
+
+The following tools were built with bazel 0.21.0 at commit 7f97ed4df69c2d9358a063cc7bb310c615ff6496 by running:
```
Comment (lberki) [inline third_party/java/java_tools/README.md]: nit: it's nicer to build all the tools at the same commit, but it's OK to do that once all of them are moved to checked-in deploy jars.

File: tools/jdk/BUILD [hunk 2]
```diff
@@ -195,6 +195,26 @@ java_import(
     neverlink = 1,
 )
 
+java_import(
+    name = "ImportDepsChecker",
+    jars = ["//tools/jdk:ImportDepsChecker_deploy.jar"]
+)
+
+java_binary(
```
Comment (lberki) [inline tools/jdk/BUILD]: Technically, this is used only in Android code, but it looks like a useful Java tool, so move on.

**Comment by iirina** [comment]
I still have second thoughts whether to push this or not, since it's only used in the Android rules... I don't think we should mix the tools (it could be problematic when we decouple Android from Bazel as well).

**Comment by lberki** [comment]
Well, what does it do? According to a Javadoc, _"Checker that checks the classes in the input jars have complete dependencies. If not, output the missing dependencies to a file"_ and it processes jdeps files and it does not contain any logic that would be specific to Android, so it makes sense to put in in tools specific to Java...

**Comment by meisterT** [comment]
Irina, did you discuss with Jin where it should belong?

**Comment by iirina** [comment]
It should belong to the android tools. I'll leave it up to Jin to come up with a plan to build and embed it in the remote android tools.

**Comment by jin** [comment]
I'll look into this alongside #1055.

**Pull request closed without merging**